expression/builtin_math.go

```diff
@@ -172,34 +175,66 @@ type floorFunctionClass struct {
 }
 
 func (c *floorFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinFloorSig{newBaseBuiltinFunc(args, ctx)}
+	if len(args) != 1 {
```

Comment:
c.verifyArgs(args) will check the len(args)
you can move it from line 209 to here.